### PR TITLE
feat: Provide scaler for Amazon managed service for Prometheus

### PR DESCRIPTION
**Pull request by sguruvar**
_Provide a description of what has been changed_

### Checklist

- [X] Commits are signed with Developer Certificate of Origin (DCO)

Fixes #
https://github.com/kedacore/keda/issues/2214

### CODE REVIEW

File: content/docs/2.13/scalers/prometheus.md [hunk 1]
```diff
@@ -90,7 +90,22 @@ See the follwowing steps to configure the scaler integration.
...
+
+- [EKS Pod Identity](https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-eks-pod-identity/) provider can be used in `authenticationRef` - see later in example. TriggerAuthentication and Secret are also supported authentication methods.
+- Create Amazon Managed Service for Prometheus [workspace](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-create-workspace.html) in your AWS account
+- Retrieve the Prometheus query endpoint URL from the [AWS managed Prometheus Workspace](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-create-workspace.html). This endpoint will be used to send queries.
+- Configure Prometheus scaler to use the workspace endpoint and an authentication method like EKS Pod Identity.
+
+Using the managed service eliminates the operational burden of running your own Prometheus servers. Queries can be executed against a fully managed, auto-scaling Prometheus deployment on AWS. Costs scale linearly with usage.
```
Comment (zroubalik) [inline content/docs/2.13/scalers/prometheus.md:105]: Could you please add an example and link it here, the similar way it is done fo Google Managed Prom. Thanks

**Review by zroubalik — CHANGES_REQUESTED** [review]

File: content/docs/2.13/scalers/prometheus.md [hunk 2]
```diff
@@ -90,7 +90,22 @@ See the follwowing steps to configure the scaler integration.
 - Prometheus server address should follow the Google's Monitoring API for [Prometheus HTTP API](https://cloud.google.com/stackdriver/docs/managed-prometheus/query#api-prometheus):
   - Example: `https://monitoring.googleapis.com/v1/projects/GOOGLE_PROJECT_ID/location/global/prometheus` - where `GOOGLE_PROJECT_ID` should be replaced by your Google project ID.
 
-To gain a better understanding of creating a Prometheus trigger for Google Managed Prometheus, refer to [this example](#example-google-managed-prometheus).
+To gain a better understanding of creating a Prometheus trigger for Google Managed Prometheus, refer to [this example]
+
+
+**Amazon Managed Service for Prometheus:**
```
Comment (tomkerkhove) [inline content/docs/2.13/scalers/prometheus.md:96]: Let's make these cloud-specific offers more discoverable by using headers:
```suggestion
### Amazon Managed Service for Prometheus
```

Please align other cloud providers as well and alphabetically sort them under a new `## Integrating Cloud offerings` section.

File: content/docs/2.13/scalers/prometheus.md [hunk 3]
```diff
@@ -90,7 +90,22 @@ See the follwowing steps to configure the scaler integration.
...
+
+- [EKS Pod Identity](https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-eks-pod-identity/) provider can be used in `authenticationRef` - see later in example. TriggerAuthentication and Secret are also supported authentication methods.
+- Create Amazon Managed Service for Prometheus [workspace](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-create-workspace.html) in your AWS account
+- Retrieve the Prometheus query endpoint URL from the [AWS managed Prometheus Workspace](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-create-workspace.html). This endpoint will be used to send queries.
+- Configure Prometheus scaler to use the workspace endpoint and an authentication method like EKS Pod Identity.
+
+Using the managed service eliminates the operational burden of running your own Prometheus servers. Queries can be executed against a fully managed, auto-scaling Prometheus deployment on AWS. Costs scale linearly with usage.
+
+
+(#example-google-managed-prometheus).
```
[the remark this answers is not in the document]
Comment (zroubalik) [reply content/docs/2.13/scalers/prometheus.md:108]: The problem is there:  this:`(#example-google-managed-prometheus).` is a part of link on line 93.

File: content/docs/2.13/scalers/prometheus.md [hunk 4]
```diff
@@ -90,7 +103,9 @@ See the follwowing steps to configure the scaler integration.
 - Prometheus server address should follow the Google's Monitoring API for [Prometheus HTTP API](https://cloud.google.com/stackdriver/docs/managed-prometheus/query#api-prometheus):
   - Example: `https://monitoring.googleapis.com/v1/projects/GOOGLE_PROJECT_ID/location/global/prometheus` - where `GOOGLE_PROJECT_ID` should be replaced by your Google project ID.
 
-To gain a better understanding of creating a Prometheus trigger for Google Managed Prometheus, refer to [this example](#example-google-managed-prometheus).
+To gain a better understanding of creating a Prometheus trigger for Google Managed Prometheus, refer to [this example]
+
+(#example-google-managed-prometheus).
```
Comment (zroubalik) [inline content/docs/2.13/scalers/prometheus.md:108]:
```suggestion
To gain a better understanding of creating a Prometheus trigger for Google Managed Prometheus, refer to [this example](#example-google-managed-prometheus).
```

File: content/docs/2.13/scalers/prometheus.md [hunk 5]
```diff
@@ -77,6 +77,19 @@ Azure has a [managed service for Prometheus](https://learn.microsoft.com/en-us/a
...
+
+- [EKS Pod Identity](https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-eks-pod-identity/) provider can be used in `authenticationRef` - see later in example. TriggerAuthentication and Secret are also supported authentication methods.
+- Create Amazon Managed Service for Prometheus [workspace](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-create-workspace.html) in your AWS account
+- Retrieve the Prometheus query endpoint URL from the [AWS managed Prometheus Workspace](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-create-workspace.html). This endpoint will be used to send queries.
+- Configure Prometheus scaler to use the workspace endpoint and an authentication method like EKS Pod Identity.
+
+Using the managed service eliminates the operational burden of running your own Prometheus servers. Queries can be executed against a fully managed, auto-scaling Prometheus deployment on AWS. Costs scale linearly with usage.
+
+
 **Google Managed for Prometheus:**
```
Comment (zroubalik) [inline content/docs/2.13/scalers/prometheus.md:93]:
```suggestion
### Google Managed for Prometheus
```

File: content/docs/2.13/scalers/prometheus.md [hunk 6]
```diff
@@ -77,7 +77,20 @@ Azure has a [managed service for Prometheus](https://learn.microsoft.com/en-us/a
...
+
+- [EKS Pod Identity](https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-eks-pod-identity/) provider can be used in `authenticationRef` - see later in example. TriggerAuthentication and Secret are also supported authentication methods.
+- Create Amazon Managed Service for Prometheus [workspace](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-create-workspace.html) in your AWS account
+- Retrieve the Prometheus query endpoint URL from the [AWS managed Prometheus Workspace](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-create-workspace.html). This endpoint will be used to send queries.
+- Configure Prometheus scaler to use the workspace endpoint and an authentication method like EKS Pod Identity.
+
+Using the managed service eliminates the operational burden of running your own Prometheus servers. Queries can be executed against a fully managed, auto-scaling Prometheus deployment on AWS. Costs scale linearly with usage.
+
+
+### Google Managed for Prometheus
```
Comment (tomkerkhove) [inline content/docs/2.13/scalers/prometheus.md:93]: Massive nit
```suggestion

### Google Managed for Prometheus
```

File: content/docs/2.13/scalers/prometheus.md (hunk 2 above)
Comment (tomkerkhove) [reply content/docs/2.13/scalers/prometheus.md]: Thanks for updating, but it looks like we did not align Azure + we did not add the new `## Integrating Cloud offerings` section for this.

**Review by tomkerkhove — CHANGES_REQUESTED** [review]

File: content/docs/2.13/scalers/prometheus.md (hunk 2 above)
Comment (sguruvar) [reply content/docs/2.13/scalers/prometheus.md]: Updated the alignment for Azure..Not sure what is meant by `## Integrating Cloud offerings`.. I already added an `#### Example: Amazon Managed Service for Prometheus (AMP)` section there and authentication provider already exists for EKS here: https://keda.sh/docs/2.11/authentication-providers/aws-eks/

Comment (zroubalik) [reply content/docs/2.13/scalers/prometheus.md]: I fixed this.

**Comment by tomkerkhove** [comment]
Perfect!

image

**Review by tomkerkhove — APPROVED** [review]

**Review by zroubalik — APPROVED** [review]

**Pull request merged**